frameworks/kafka/src/main/java/com/mesosphere/sdk/kafka/scheduler/KafkaService.java

```diff
@@ -20,22 +19,31 @@
  */
 public class KafkaService extends DefaultService {
     protected static final Logger LOGGER = LoggerFactory.getLogger(KafkaService.class);
-
+    final KafkaZKClient kafkaZKClient;
 
     public KafkaService(File pathToYamlSpecification) throws Exception {
-        super(pathToYamlSpecification);
+        RawServiceSpec rawServiceSpec = YAMLServiceSpecFactory.generateRawSpecFromYAML(pathToYamlSpecification);
+        DefaultScheduler.Builder schedulerBuilder =
+                DefaultScheduler.newBuilder(YAMLServiceSpecFactory.generateServiceSpec(rawServiceSpec));
+        schedulerBuilder.setPlansFrom(rawServiceSpec);
+
+        kafkaZKClient = new KafkaZKClient(schedulerBuilder.getServiceSpec().getZookeeperConnection(),
+                DcosConstants.SERVICE_ROOT_PATH_PREFIX + schedulerBuilder.getServiceSpec().getName());
+
+        schedulerBuilder.setEndpointProducer("address", new BrokerAddress(kafkaZKClient));
```

Comment:
`address` isn't needed. Broker endpoints are already being provided automatically from the `broker` VIP information:
```
$ dcos kafka endpoints
[
  "broker"
]
$ dcos kafka endpoints broker
{
  "direct": [
    "kafka-2-broker.kafka.mesos:1025",
    "kafka-0-broker.kafka.mesos:1025",
    "kafka-1-broker.kafka.mesos:1025"
  ],
  "vip": "broker.kafka.l4lb.thisdcos.directory:9092"
}
```